src/main/java/io/strimzi/kafka/quotas/StorageChecker.java

```diff
@@ -77,13 +81,25 @@ public void run() {
         }
     }
 
+    @Deprecated(forRemoval = true, since = "0.3.0")
     long checkDiskUsage() {
+        return totalDiskUSage(gatherDiskUsage());
+    }
+
+    long totalDiskUSage(Map<String, Long> diskUsage) {
+        return diskUsage.values().stream().mapToLong(Long::longValue).sum();
+    }
+
+    Map<String, Long> gatherDiskUsage() {
         return logDirs.stream()
                 .filter(Files::exists)
                 .map(path -> apply(() -> Files.getFileStore(path)))
                 .distinct()
-                .mapToLong(store -> apply(() -> store.getTotalSpace() - store.getUsableSpace()))
-                .sum();
+                .map(store -> {
+                    final Long usedSpace = apply(() -> store.getTotalSpace() - store.getUsableSpace());
+                    return new AbstractMap.SimpleImmutableEntry<>(store.name(), usedSpace != null ? usedSpace : 0);
+                })
+                .collect(Collectors.toMap(AbstractMap.SimpleImmutableEntry::getKey, AbstractMap.SimpleImmutableEntry::getValue));
```

Comment:
I think you can say the following and achieve the same (untested).
```suggestion
.collect(Collectors.toMap(FileStore::name, StorageChecker::calculateUsedSpace));
```

where
```
    private static long calculateUsedSpace(FileStore store) {
        Long usedSpace = apply(() -> store.getTotalSpace() - store.getUsableSpace());
        return usedSpace != null ? usedSpace : 0;
    }
```

Reply:
Ohh nice.